packages/format-library/src/code/index.js

```diff
@@ -20,16 +20,18 @@ export const code = {
 	__unstableInputRule( value ) {
 		const BACKTICK = '`';
 		const { start, text } = value;
-		const characterBefore = text.slice( start - 1, start );
+		const characterBefore = text[ start - 1 ];
 
 		// Quick check the text for the necessary character.
 		if ( characterBefore !== BACKTICK ) {
 			return value;
 		}
 
-		const textBefore = text.slice( 0, start - 1 );
-		const indexBefore = textBefore.lastIndexOf( BACKTICK );
+		if ( start - 2 < 0 ) {
```

Comment:
When `start` is  0 then `characterBefore` will always be `undefined`. When `start` is 1 then this will also bail out so we don't care so much about `characterBefore`. Maybe this check should run before line 23?

Reply:
I don't understand what is happening here.

Reply:
my purpose here is to minimize the change to leave it as small and focused as possible. what's happening is exactly what @gziolo noted. the check ensures that we don't introduce a regression by moving to `lastIndexOf`, which treats its `position` parameter as `0` if it's negative. without the `- 2` check we would conflate backticks at `0` with no previous backticks.

Reply:
Would be good to comment inline